optuna/samplers/_gp/sampler.py

```diff
@@ -329,14 +329,15 @@ def _warn_and_convert_inf(
 
 def _get_constraint_vals_and_feasibility(
     study: Study, trials: list[FrozenTrial]
-) -> tuple[np.ndarray, np.bool | np.ndarray]:
```

Comment:
The second return is definitely not boolean, so I removed this to avoid any confusion.